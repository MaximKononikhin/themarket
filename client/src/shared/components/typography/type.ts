import { ReactNode } from "react";

export const TYPOGRAPHY = [
    "header-1",
    "header-2",
    "text-1",
    "text-1-medium",
    "text-2",
    "text-2-medium",
    "text-3",
] as const;

export type Type = (typeof TYPOGRAPHY)[number];

export type TypographyProps = {
    /** Название стилей по макету ***/
    type: Type;
    /** Потом ***/
    children: ReactNode;
    /** Доп.Стили ***/
    className?: string;
    /** test **/
    testId?: string;
};
