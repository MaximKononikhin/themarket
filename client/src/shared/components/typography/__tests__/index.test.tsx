import React from "react";

import { render } from "@testing-library/react";

import { Typography } from "../index";
import { TYPOGRAPHY } from "../type";

describe("Typography component", () => {
    TYPOGRAPHY.map((type) => {
        it(`should render component with type ${type}`, () => {
            const testId = `test_${type}`;
            const { getByTestId, container } = render(
                <Typography testId={testId} type={type}>
                    {type}
                </Typography>
            );
            expect(getByTestId(testId)).toBeInTheDocument();
            expect(container).toMatchSnapshot();
        });
    });
});
