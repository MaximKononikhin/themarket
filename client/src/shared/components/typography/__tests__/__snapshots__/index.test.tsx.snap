// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component should render component with type error 1`] = `
<div>
  <p
    class="textError"
    data-testid="test_error"
  >
    error
  </p>
</div>
`;

exports[`Typography component should render component with type header-1 1`] = `
<div>
  <h1
    class="heading1"
    data-testid="test_header-1"
  >
    header-1
  </h1>
</div>
`;

exports[`Typography component should render component with type header-2 1`] = `
<div>
  <h2
    class="heading2"
    data-testid="test_header-2"
  >
    header-2
  </h2>
</div>
`;

exports[`Typography component should render component with type text-1 1`] = `
<div>
  <p
    class="paragraphOne"
    data-testid="test_text-1"
  >
    text-1
  </p>
</div>
`;

exports[`Typography component should render component with type text-1-medium 1`] = `
<div>
  <p
    class="textOneMedium"
    data-testid="test_text-1-medium"
  >
    text-1-medium
  </p>
</div>
`;

exports[`Typography component should render component with type text-2 1`] = `
<div>
  <p
    class="textSecond"
    data-testid="test_text-2"
  >
    text-2
  </p>
</div>
`;

exports[`Typography component should render component with type text-2-medium 1`] = `
<div>
  <p
    class="textTwoMedium"
    data-testid="test_text-2-medium"
  >
    text-2-medium
  </p>
</div>
`;

exports[`Typography component should render component with type text-3 1`] = `
<div>
  <p
    class="textThird"
    data-testid="test_text-3"
  >
    text-3
  </p>
</div>
`;
